# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/cairo/cairo/src/cairo-xcb-connection-core.c
# Commit: 3cdfb39376b7
# Full Hash: 3cdfb39376b736597fe658a322389117686b7e82
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2024-05-07 09:13:07
# Regressor Bug: 1892913
# File Overlap Count: 3
# Description:
#   Bug 1892913 - patch 1 - Update cairo source to 1.18.0 r=gfx-reviewers,lsalzman
#   
#   Wholesale replacement of files in gfx/cairo/cairo with contents of the 1.18.0 release tarball,
#   omitting subdirectories that are not part of the core library build.
#   
# ==============================================================================

diff -r 9daa351f875b -r 3cdfb39376b7 gfx/cairo/cairo/src/cairo-xcb-connection-core.c
--- a/gfx/cairo/cairo/src/cairo-xcb-connection-core.c	Mon May 06 19:22:35 2024 +0000
+++ b/gfx/cairo/cairo/src/cairo-xcb-connection-core.c	Mon May 06 19:25:12 2024 +0000
@@ -42,7 +42,7 @@
 				     uint16_t width,
 				     uint16_t height)
 {
-    xcb_pixmap_t pixmap = _cairo_xcb_connection_get_xid (connection);
+    xcb_pixmap_t pixmap = xcb_generate_id (connection->xcb_connection);
 
     assert (width > 0);
     assert (height > 0);
@@ -52,35 +52,19 @@
     return pixmap;
 }
 
-void
-_cairo_xcb_connection_free_pixmap (cairo_xcb_connection_t *connection,
-				   xcb_pixmap_t pixmap)
-{
-    xcb_free_pixmap (connection->xcb_connection, pixmap);
-    _cairo_xcb_connection_put_xid (connection, pixmap);
-}
-
 xcb_gcontext_t
 _cairo_xcb_connection_create_gc (cairo_xcb_connection_t *connection,
 				 xcb_drawable_t drawable,
 				 uint32_t value_mask,
 				 uint32_t *values)
 {
-    xcb_gcontext_t gc = _cairo_xcb_connection_get_xid (connection);
+    xcb_gcontext_t gc = xcb_generate_id (connection->xcb_connection);
     xcb_create_gc (connection->xcb_connection, gc, drawable,
 		   value_mask, values);
     return gc;
 }
 
 void
-_cairo_xcb_connection_free_gc (cairo_xcb_connection_t *connection,
-			       xcb_gcontext_t gc)
-{
-    xcb_free_gc (connection->xcb_connection, gc);
-    _cairo_xcb_connection_put_xid (connection, gc);
-}
-
-void
 _cairo_xcb_connection_change_gc (cairo_xcb_connection_t *connection,
 				 xcb_gcontext_t gc,
 				 uint32_t value_mask,